tests/page/interception.spec.ts

```diff
@@ -90,21 +89,50 @@ it('should work with glob', async () => {
   expect(globToRegex('http://localhost:3000/signin-oidc*').test('http://localhost:3000/signin-oidc/foo')).toBeFalsy();
   expect(globToRegex('http://localhost:3000/signin-oidc*').test('http://localhost:3000/signin-oidcnice')).toBeTruthy();
 
-  // range []
-  expect(globToRegex('**/api/v[0-9]').test('http://example.com/api/v1')).toBeTruthy();
+  // range [] is NOT supported
+  expect(globToRegex('**/api/v[0-9]').test('http://example.com/api/v[0-9]')).toBeTruthy();
   expect(globToRegex('**/api/v[0-9]').test('http://example.com/api/version')).toBeFalsy();
 
   // query params
   expect(globToRegex('**/api\\?param').test('http://example.com/api?param')).toBeTruthy();
   expect(globToRegex('**/api\\?param').test('http://example.com/api-param')).toBeFalsy();
-  expect(globToRegex('**/three-columns/settings.html\\?**id=[a-z]**').test('http://mydomain:8080/blah/blah/three-columns/settings.html?id=settings-e3c58efe-02e9-44b0-97ac-dd138100cf7c&blah')).toBeTruthy();
+  expect(globToRegex('**/three-columns/settings.html\\?**id=settings-**').test('http://mydomain:8080/blah/blah/three-columns/settings.html?id=settings-e3c58efe-02e9-44b0-97ac-dd138100cf7c&blah')).toBeTruthy();
 
   expect(globToRegex('\\?')).toEqual(/^\?$/);
   expect(globToRegex('\\')).toEqual(/^\\$/);
   expect(globToRegex('\\\\')).toEqual(/^\\$/);
   expect(globToRegex('\\[')).toEqual(/^\[$/);
-  expect(globToRegex('[a-z]')).toEqual(/^[a-z]$/);
+  expect(globToRegex('[a-z]')).toEqual(/^\[a-z\]$/);
   expect(globToRegex('$^+.\\*()|\\?\\{\\}\\[\\]')).toEqual(/^\$\^\+\.\*\(\)\|\?\{\}\[\]$/);
+
+  expect(urlMatches(undefined, 'http://playwright.dev/', 'http://playwright.dev')).toBeTruthy();
```

Comment:
I don't see a test for escaped ? (we've seen a bunch in the wild). Also not seeing where you support it.

Reply:
> I don't see a test for escaped ? (we've seen a bunch in the wild).

There is a few tests in `interception.spec.ts` and also in `tests/page/page-route.spec.ts`. Escaped behavior didn't change.

> Also not seeing where you support it.

See changes in `urlMatcher.ts`

Reply:
I can't see it there!